packages/contracts-bedrock/test/vendor/Initializable.t.sol

```diff
@@ -352,7 +352,8 @@ contract Initializer_Test is Bridge_Initializer {
         // Ensure that all L1, L2 `Initializable` contracts are accounted for, in addition to
         // OptimismMintableERC20FactoryImpl, OptimismMintableERC20FactoryProxy, OptimismPortal2,
         // DisputeGameFactoryImpl, DisputeGameFactoryProxy, DelayedWETHImpl, DelayedWETHProxy.
-        assertEq(_getNumInitializable() + 1, contracts.length);
+        // Omitting OptimismSuperchainERC20 due to using OZ v5 Initializable.
+        assertEq(_getNumInitializable(), contracts.length);
```

Comment:
We need to find a better way around this, omitting a contract in a test like this can lead to bugs/confusions. I know @tynes had the idea of slowly moving to unstructured storage which the modern OZ provides (possibly moving to a custom initializer is the best option), so other contracts may have this problem of having a different error/compiler version in the future. 

Perhaps we can make a new test file like this one for the newer contracts that uses pragma `0.8.25`, or see if we can include a `version` field to the struct and assert the error based on that, but this won't fix the invalid compilers versions.

Reply:
Could we add a new test in this file for the new layout?

Reply:
The thing is that in this file we can't create the `OptimismSuperchainERC20` because it is using `0.8.25`. We can create another file for the OZ v5 Initializable contracts called `InitializerOZv5.t.sol`.

This is also useful due to that the current tests uses the artifacts for checking the `storageLayout` snapshots but in the OZ v5 implementation that wont work with the namespaces.

Reply:
We will need to come back and pay off this tech debt eventually but its ok for now